aea/components/base.py

```diff
@@ -339,15 +339,22 @@ def perform_load_aea_package(
 
     prefix_root_module = types.ModuleType(prefix_root)
     prefix_root_module.__path__ = None  # type: ignore
-    sys.modules[prefix_root] = sys.modules.get(prefix_root, prefix_root_module)
+    actual_prefix_root_module = sys.modules.get(prefix_root, prefix_root_module)
+    sys.modules[prefix_root] = actual_prefix_root_module
     author_module = types.ModuleType(prefix_author)
     author_module.__path__ = None  # type: ignore
-    sys.modules[prefix_author] = sys.modules.get(prefix_author, author_module)
+    actual_prefix_author_module = sys.modules.get(prefix_author, author_module)
+    sys.modules[prefix_author] = actual_prefix_author_module
+    setattr(actual_prefix_root_module, author, actual_prefix_author_module)
```

Comment:
@marcofavorito can you explain

Reply:
Please put in context of this (https://github.com/valory-xyz/open-aea/pull/273#issuecomment-1229504771) :)

Reply:
This code snippet shows the behaviour of the Python import system wrt how `sys.modules` is populated: 

```python
import sys
from copy import copy
import aea
old_sys_modules = copy(sys.modules)

import packages.valory.connections.p2p_libp2p
new_sys_modules = copy(sys.modules)

new_imported_modules = set(new_sys_modules.keys()) - set(old_sys_modules.keys())
print(new_imported_modules)
```

It prints:
```python
{'packages', 'packages.valory', 'packages.valory.connections.p2p_libp2p', 'packages.valory.connections'}
```

Crucially, the modules corresponding to the printed keys of `sys.modules` can access to the children modules by means of attribute getters:
```python
packages_module = sys.modules["packages"]
packages_valory_module = sys.modules["packages.valory"]
packages_valory_connections_module = sys.modules["packages.valory.connections"]
packages_valory_connections_p2p_libp2p_module = sys.modules["packages.valory.connections.p2p_libp2p"]

# the following defined because the submodule is set as an attribute of the parent module
packages_module.valory
assert packages_module.valory == packages_valory_module
assert packages_valory_connections_module == packages.valory.connections
assert packages_valory_connections_p2p_libp2p_module == packages.valory.connections.p2p_libp2p
```

Before commit https://github.com/valory-xyz/open-aea/commit/5243e1eecda1ca10ed7608bee895f89633fa7da4 we did not comply with the standard behaviour; in particular, in the case when the module attribute `__path__` is not `None`, the following should happen: 

![image](https://user-images.githubusercontent.com/9467617/188308369-97ab6393-d646-40be-ae78-43bfee369829.png)

There is a little caveat: `packages` is a virtual module in our system, as in an execution of an AEA project it does not distinguish between vendor and non-vendor packages, and so `__path__ = None` (as it does not correspond to any `__init__.py` file. The same can be said for `packages.author` packages for non-vendor packages as there is no `__init__.py` file that makes the package mappable to the filesystem.

So, instead of having to specify the whole sequence in the import statement:

```python
from packages.author.skills.skill_name import that_module
```

One could do:

```python
import packages.valory.protocols.acn
import packages.valory.connections.p2p_libp2p

# now this is a valid access to the module:
packages.valory
```

In particular, in our tests we often access to a package module by its full import path, as shown in the comment above:
```python
@mock.patch.object(
    packages.fetchai.protocols.contract_api.message,
    "enforce",
    side_effect=AEAEnforceError("some error"),
)
...
```